### DISCUSSION TITLE
How do I import the validators in src/dsl/validator?

### DISCUSSION THREAD

**Discussion by ally-ahmed** [instructor-ai/instructor-js#121] [Q&A] [answered]
The validators are not exported in `src/instructor.ts` so if we want to use the the validators defined in `src/dsl/validator.ts` we can't. 
The examples [Self-Correction with LLMValidator ](https://instructor-ai.github.io/instructor-js/examples/self_correction/#self-correction-with-llmvalidator) and [OpenAI Moderation](https://instructor-ai.github.io/instructor-js/examples/content_moderation/#openai-moderation) both suggest that import from `@/dsl/validator` should work but it doesn't. 

Is it possible to use the validators or do we need to copy paste them into our code base or maybe it was an oversight and a pull request to export the validators in `src/instructor.ts` is required?

**Comment by c-emman** [comment]
I am in the same position as yourself I can't import also. How did you get around this? Did you just copy and paste the code?

**Reply by ally-ahmed** [reply]
@c-emman I was just experimenting with the library at the time, I didn't go ahead with trying to copy paste the validators and see if that works. If you get around trying it out let me know if it works.

**Reply by c-emman** [reply]
I managed to get it working by copying and pasting it. But having trouble with page refreshes when i use it. I want it to be an api call that runs all the retries and returns the result at the end but some weird behaviour, Im using nextjs so maybe that has something to do with it

**Comment by jxnl** [comment, marked as answer]
https://discord.com/channels/1192334452110659664/1192336743496683580/1242950190609203241
